### ISSUE TITLE
circleci: cache the docker-spy image

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#3025]
The following test appears to have failed:

[#8991](https://circleci.com/gh/cockroachdb/cockroach/8991):

```
=== RUN   TestBuildInfo
I1105 23:07:19.511778 15932 acceptance/localcluster/localcluster.go:196  pulling cockroachdb/docker-spy
panic: test timed out after 5m0s

goroutine 29 [running]:
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:703 +0x132
created by time.goFunc
	/usr/local/go/src/time/sleep.go:129 +0x3a

goroutine 1 [chan receive]:
testing.RunTests(0xb5e5b0, 0xf2d1c0, 0x8, 0x8, 0xa59301)
	/usr/local/go/src/testing/testing.go:562 +0x8ad

```
Please assign, take a look and update the issue accordingly.

**Comment by tamird** [comment]
We're not currently caching the docker-spy image, so this happens every now and again. Bummer.

**Comment by luuhaidang** [comment]
Chip-a9

**Comment by luuhaidang** [comment]
=== RUN   TestBuildInfo
I1105 23:07:19.511778 15932 acceptance/localcluster/localcluster.go:196  pulling cockroachdb/docker-spy
panic: test timed out after 5m0s

goroutine 29 [running]:
testing.startAlarm.func1()
    /usr/local/go/src/testing/testing.go:703 +0x132
created by time.goFunc
    /usr/local/go/src/time/sleep.go:129 +0x3a

goroutine 1 [chan receive]:
testing.RunTests(0xb5e5b0, 0xf2d1c0, 0x8, 0x8, 0xa59301)
    /usr/local/go/src/testing/testing.go:562 +0x8ad

**Comment by petermattis** [comment]
@mjibson Should be straightforward as part of the other CircleCI docker image caching you were doing.

**Comment by petermattis** [comment]
Fixed by #4414.
